packages/block-library/src/image/image.js

```diff
@@ -480,15 +480,27 @@ export default function Image( {
 					{ isResizable && (
 						<DimensionsTool
 							value={ { width, height, scale, aspectRatio } }
-							onChange={ ( newValue ) => {
+							onChange={ ( {
+								width: newWidth,
+								height: newHeight,
+								scale: newScale,
+								aspectRatio: newAspectRatio,
+							} ) => {
 								// Rebuilding the object forces setting `undefined`
 								// for values that are removed since setAttributes
 								// doesn't do anything with keys that aren't set.
 								setAttributes( {
-									width: newValue.width,
-									height: newValue.height,
-									scale: newValue.scale,
-									aspectRatio: newValue.aspectRatio,
+									// CSS includes `height: auto`, but we need
+									// `width: auto` to fix the aspect ratio when
+									// only height is set due to the width and
+									// height attributes set via the server.
+									width:
+										! newWidth && newHeight
+											? 'auto'
+											: newWidth,
```

Comment:
It was already a really long comment, but I couldn't add `height: auto` to the CSS either because it breaks aspect ratio when it and `width: auto` are both set.

I also can't really remove `height: auto` from the CSS or it'll break existing blocks.